project/src/main/puzzle/pickups.gd

```diff
@@ -89,6 +96,24 @@ func clear() -> void:
 	_pickups_by_cell.clear()
 
 
+func set_piece_manager_path(new_piece_manager_path: NodePath) -> void:
+	piece_manager_path = new_piece_manager_path
+	_refresh_piece_manager_path()
+
+
+func _refresh_piece_manager_path() -> void:
+	if not is_inside_tree():
+		return
+	
+	if _piece_manager:
+		_piece_manager.disconnect("piece_disturbed", self, "_on_PieceManager_piece_disturbed")
+	
+	_piece_manager = get_node(piece_manager_path) if piece_manager_path else null
+	
+	if _piece_manager:
+		_piece_manager.connect("piece_disturbed", self, "_on_PieceManager_piece_disturbed")
```

Comment:
You are connecting a non-existent signal.

Reply:
The signal receiver method already existed and was not shown in this diff, but it exists.